src/mo_interpreter/interpret.ml

```diff
@@ -705,43 +705,43 @@ and declare_pat_fields pfs ve : val_env =
 and define_id env id v =
   Lib.Promise.fulfill (find id.it env.vals) v
 
-and define_pat env pat v =
-  let err () = trap pat.at "value %s does not match pattern" (string_of_val env v) in
-  define_pat' env err pat v
-
-and define_pat' env err pat v =
+and define_pat env (k : V.value V.cont) err pat v =
```

Comment:
maybe we could have a `type continuation = V.(value cont);` in a cleanup PR sometime.